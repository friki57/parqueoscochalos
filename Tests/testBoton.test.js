import React from 'react';
import ReactDom from 'react-dom';
import Boton from './../src/Vista/desarrollo/componentes/boton.js'
import SuperGeneradorFormularios3000 from './../src/Vista/desarrollo/componentes/superGeneradorFormularios3000.js'
import {render,cleanup} from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from "react-test-renderer";

// it("Jala el botoncito", ()=>
// {
//   const {getByTestId} = render(<Boton texto = 'apreta'/>)
//   expect(getByTestId('boton')).toHaveTextContent("apreta");
// });
afterEach(cleanup);

it("Test 1", ()=>{
  const tree= renderer.create(<Boton texto = "hola"/>).toJSON();
  expect(tree).toMatchSnapshot();
  });

visit("Super form 1", ()=>{
  window.datos = {
    SuperGeneradorFormularios3000:
      {
        titulo: "Iniciar sesión",
        method: 'post',
        action: '/inicioSesion',
        campos:
        [
          {
            name: 'correo',
            placeholder: 'Dirección de correo electrónico',
            value: '',
            label: 'Correo Electrónico: ',
            type: 'email'
          },
          {
            name: 'contra',
            placeholder: 'Contraseña',
            value: '',
            label: 'Contraseña',
            type: 'password'
          }
        ]
      }
    }


  const tree= renderer.create(<SuperGeneradorFormularios3000/>).toJSON();
  expect(tree).toMatchSnapshot();
  });
