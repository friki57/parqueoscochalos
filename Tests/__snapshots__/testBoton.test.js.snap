// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Super form 1 1`] = `
<div>
  <form
    action="/inicioSesion"
    method="post"
  >
    <h1
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      Iniciar sesión
    </h1>
    <div
      style={
        Object {
          "border": "none",
          "color": "#fff",
          "textAlign": "left",
          "width": "360pt",
        }
      }
    >
      <label
        htmlFor="correo"
      >
         
        Correo Electrónico: 
         
      </label>
      <br />
      <input
        defaultValue=""
        name="correo"
        placeholder="Dirección de correo electrónico"
        style={
          Object {
            "background": "#fff6",
            "border": "none",
            "borderRadius": "15px",
            "color": "#000",
            "display": "block",
            "padding": "10px",
            "width": "100%",
          }
        }
        type="email"
      />
      <br />
    </div>
    <div
      style={
        Object {
          "border": "none",
          "color": "#fff",
          "textAlign": "left",
          "width": "360pt",
        }
      }
    >
      <label
        htmlFor="contra"
      >
         
        Contraseña
         
      </label>
      <br />
      <input
        defaultValue=""
        name="contra"
        placeholder="Contraseña"
        style={
          Object {
            "background": "#fff6",
            "border": "none",
            "borderRadius": "15px",
            "color": "#000",
            "display": "block",
            "padding": "10px",
            "width": "100%",
          }
        }
        type="password"
      />
      <br />
    </div>
    <button
      style={
        Object {
          "background": "#D53F3D",
          "border": "none",
          "color": "#fff",
          "padding": "10px 50px",
          "textTransform": "uppercase",
          "width": "146pt",
        }
      }
      type="Submit"
    >
       Confirmar
    </button>
  </form>
</div>
`;

exports[`Test 1 1`] = `
<button
  data-testid="boton"
>
  hola
</button>
`;
